### PR TITLE
[MXNET-13453] [Clojure] - Add Spec Validations to the Optimizer namespace

### PR DESCRIPTION
**Pull request by hellonico**
## Description ##

add some spec validations to the optimizer namespace

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ x] Changes are complete (i.e. I finished coding on this PR)
- [ x] All changes have test coverage:

### Changes ###
- updated optimizer.clj with specs
- added throw exception checks in optimizer_test.clj

### CODE REVIEW

**Comment by gigasquid** [comment]
Thanks @hellonico so much for your contribution! 😸 I'll take a closer look tomorrow. Don't worry about the failure on the Travis side, there have been some issues with it lately.

**Comment by vandanavk** [comment]
@mxnet-label-bot add [Clojure, pr-awaiting-review]

File: contrib/clojure-package/test/org/apache/clojure_mxnet/optimizer_test.clj [hunk 1]
```diff
@@ -44,3 +44,13 @@
               ["sgld" optimizer/sgld]]]
     (doseq [opt opts]
       (test-optimizer opt))))
+
+(deftest test-optimizers-parameters-specs
```
Comment (gigasquid) [inline contrib/clojure-package/test/org/apache/clojure_mxnet/optimizer_test.clj]: Nice tests 👍

**Comment by gigasquid** [comment]
I kicked off the Jenkins CI again - when it passes this is good to merge.

**Comment by gigasquid** [comment]
@hellonico - Can you change the PR to be against master ? Right now it against 1.3.0 - thanks!

**Comment by gigasquid** [comment]
@hellonico I was able to change the base in the PR, but I think you will need to rebase your commits against master to make it all clean :)

**Comment by gigasquid** [comment]
Thanks @hellonico !!

**Pull request merged**